packages/app-store/_utils/writeAppDataToEventType.ts

```diff
@@ -0,0 +1,57 @@
+import { EventTypeRepository } from "@calcom/lib/server/repository/eventType";
+import prisma from "@calcom/prisma";
+import type { AppCategories } from "@calcom/prisma/enums";
+import { EventTypeMetaDataSchema } from "@calcom/prisma/zod-utils";
+
+import type { appDataSchemas } from "../apps.schemas.generated";
+
+const writeAppDataToEventType = async ({
+  userId,
+  teamId,
+  appSlug,
+  appCategories,
+  credentialId,
+}: {
+  userId?: number;
+  teamId?: number;
+  appSlug: string;
+  appCategories: AppCategories[];
+  credentialId: number;
+}) => {
+  //   Search for event types belonging to the user / team
+  const eventTypes = teamId
+    ? await EventTypeRepository.findAllByTeamIdIncludeManagedEventTypes({ teamId })
+    : userId
+    ? await EventTypeRepository.findAllByUserId({ userId })
```

Comment:
@hariombalhara I've created two methods to the `EventTypeRepository` it feels like once we start adding "or" to the method name we should split it into two separate methods.